### ISSUE TITLE
False positive from cmp_owned?

### ISSUE THREAD

**Issue by djc** [rust-lang/rust-clippy#7365]
Lint name:

cmp_owned

I tried this code:

```rust
if OrderedFloat::from(dist) < OrderedFloat::from(val.0) {
    val.0 = dist;
}
```

I expected to see this happen: nothing.

Instead, this happened:

```rust
warning: this creates an owned instance just for comparison
  --> vectors/src/generator.rs:59:16
   |
59 |             if OrderedFloat::from(dist) < OrderedFloat::from(val.0) {
   |                ^^^^^^^^^^^^^^^^^^^^^^^^----------------------------
   |                |
   |                help: try: `OrderedFloat::from(val.0) == dist`
   |
   = note: `#[warn(clippy::cmp_owned)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#cmp_owned
```

First, it's weird that the comparison operator changes from `<` to `==`. Second, the suggestion doesn't even work because `OrderedFloat` doesn't implement `Ord` with `f32` itself. Third, since `OrderedFloat` is `Copy` clippy should probably not warn about creating an owned value of that type anyway.

### Meta

- `cargo clippy -V`: e.g. clippy 0.1.53 (53cb7b09 2021-06-17)
- `rustc -Vv`:
```
rustc 1.53.0 (53cb7b09b 2021-06-17)
binary: rustc
commit-hash: 53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b
commit-date: 2021-06-17
host: x86_64-apple-darwin
release: 1.53.0
LLVM version: 12.0.1
```
